ambry-frontend/src/main/java/com/github/ambry/frontend/DeleteBlobHandler.java

```diff
@@ -148,6 +156,15 @@ private Callback<Void> securityPostProcessRequestCallback(BlobId blobId) {
      */
     private Callback<Void> routerCallback() {
       return buildCallback(metrics.deleteBlobRouterMetrics, result -> {
+        if (RestUtils.isDatasetVersionUpload(restRequest.getArgs())) {
```

Comment:
now that this header is used by many http methods, can we rename this header to something else?

Reply:
Rename to x-ambry-dataset-version-query-enabled.